src/movegen.cpp

```diff
@@ -31,7 +31,7 @@ namespace {
     if (Type == CAPTURES || Type == EVASIONS || Type == NON_EVASIONS)
```

Comment:
```suggestion
    if constexpr (Type == CAPTURES || Type == EVASIONS || Type == NON_EVASIONS)
```

Reply:
using if constexpr in two of those cases will cause the checks to fail because they trigger warnings for unused variables. for the one in movegen.cpp on line 31, you could circumvent this by doing something like 
if(type == CAPTURES) 

else if(type == QUIETS) 

else // EVASIONS, NON_EVASIONS

Let me know if something like that is preferable, or if I should leave it as-is. The suggested change in movepick line 161 should not create any problems.

Reply:
Ah, I was testing with clang++, which didn't report any warning. Just leave them as-is in the two cases then.

Reply:
See [this commit](https://github.com/official-stockfish/Stockfish/commit/5a871e174f22894837c2363b5c215854ee155113), you can use the same syntax to prevent "unused" warnings

Reply:
done!